bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/QvContext.scala

```diff
@@ -45,11 +46,11 @@ case class QvContext(peer: Peer, dataSource: DataSource, source: ObservationProv
   def timezone = site.timezone()
 
   // ===== Data providers / filters used in this context ====
-  private val _nonSidProvider = new PositionProvider(this, source)
-  private val _mainFilter = FilterProvider(_nonSidProvider)
-  private val _tableFilter = FilterProvider(_mainFilter)
-  private val _selection = SelectionProvider(_tableFilter)
-  
+  private lazy val _nonSidProvider = new PositionProvider(this, source)
+  private lazy val _mainFilter = FilterProvider(this, _nonSidProvider)
+  private lazy val _tableFilter = FilterProvider(this, _mainFilter)
+  private lazy val _selection = SelectionProvider(this, _tableFilter)
```

Comment:
`lazy` is necessary here because the constructors actually do things with the `QvContext` which is not yet initialized (since we're passing `this`). It NPEs if we don't do this.